diff --git a/src/libsodium/crypto_core/ed25519/ref10/ed25519_ref10.c b/src/libsodium/crypto_core/ed25519/ref10/ed25519_ref10.c
index dda67360..3a49dacc 100644
--- a/src/libsodium/crypto_core/ed25519/ref10/ed25519_ref10.c
+++ b/src/libsodium/crypto_core/ed25519/ref10/ed25519_ref10.c
@@ -590,13 +590,23 @@ ge25519_cmov8(ge25519_precomp *t, const ge25519_precomp precomp[8], const signed
 static void
 ge25519_cmov8_base(ge25519_precomp *t, const int pos, const signed char b)
 {
-    static const ge25519_precomp base[32][8] = { /* base[i][j] = (j+1)*256^i*B */
+#ifdef ESP8266
+#define GE25519_BASE_ATTR __attribute__((section(".irom.text")))
+// NOTE: no need to use special _P functions because all data in ge25519_precomp
+// is already word-sized
+#else
+#define GE25519_BASE_ATTR
+#endif
+    CRYPTO_ALIGN(16)
+    static const ge25519_precomp GE25519_BASE_ATTR base[32][8] = { /* base[i][j] = (j+1)*256^i*B */
 #ifdef HAVE_TI_MODE
 # include "fe_51/base.h"
 #else
 # include "fe_25_5/base.h"
 #endif
     };
+
+#undef GE25519_BASE_ATTR
     ge25519_cmov8(t, base[pos], b);
 }
 
